tests/unit/source_code/python/test_python_ast.py

```diff
@@ -194,28 +201,25 @@ def test_is_instance_of(source, name, class_name) -> None:
     assert Tree(var[0]).is_instance_of(class_name)
 
 
-def test_supports_recursive_refs_when_checking_module() -> None:
-    source_1 = """
-    df = spark.read.csv("hi")
-    """
-    source_2 = """
-    df = df.withColumn(stuff)
-    """
-    source_3 = """
-    df = df.withColumn(stuff2)
-    """
-    maybe_tree = Tree.maybe_normalized_parse(source_1)
-    assert maybe_tree.tree is not None, maybe_tree.failure
-    main_tree = maybe_tree.tree
-    maybe_tree_2 = Tree.maybe_normalized_parse(source_2)
-    assert maybe_tree_2.tree is not None, maybe_tree_2.failure
-    tree_2 = maybe_tree_2.tree
-    main_tree.append_tree(tree_2)
-    maybe_tree_3 = Tree.maybe_normalized_parse(source_3)
-    assert maybe_tree_3.tree is not None, maybe_tree_3.failure
-    tree_3 = maybe_tree_3.tree
-    main_tree.append_tree(tree_3)
-    assign = tree_3.locate(Assign, [])[0]
+def test_tree_attach_child_tree_propagates_module_reference() -> None:
+    """The spark module should propagate from the parent tree."""
+    sources = (
+        "df = spark.read.csv('hi')",
+        "df = df.withColumn(stuff)",
+        "df = df.withColumn(stuff2)",
+    )
```

Comment:
I think the original newlines should be preserved?

The real smell here is the array indexing below. I thought about destructuring assignment:
```suggestion
    (source_1, source_2, source_3) = (
        "df = spark.read.csv('hi')\n",
        "df = df.withColumn(stuff)\n",
        "df = df.withColumn(stuff2)\n",
    )
```
… but at that point I think it's best to just stick with something closer to the original:
```suggestion
    source_1 = "df = spark.read.csv('hi')\n"
    source_2 = "df = df.withColumn(stuff)\n"
    source_3 = "df = df.withColumn(stuff2)\n"
```

This is all definitely in the category of style nits rather than anything important.